manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/MatchTest.scala

```diff
@@ -19,347 +19,305 @@
-  }
-
-  @Test def get_all_nodes() {
-    testQuery(
-      title = "Get all nodes",
-      text = "By just specifying a pattern with a single node and no labels, all nodes in the graph will be returned.",
-      queryText = """match (n) return n""",
-      optionalResultExplanation = "Returns all the nodes in the database.",
-      assertions = (p) => {
-        val allNodes: List[Node] = GlobalGraphOperations.at(db).getAllNodes.asScala.toList
-
-        assertEquals(allNodes, p.columnAs[Node]("n").toList)
-      }
+class MatchTest extends DocumentingTest {
+  override def outputPath = "target/docs/dev/ql/"
+  override def doc = new DocBuilder {
+    doc("Match", "query-match")
+    initQueries(
+      """CREATE (charlie:Person {name:'Charlie Sheen'}),
+        |       (martin:Person {name: 'Martin Sheen'}),
+        |       (michael:Person {name: 'Michael Douglas'}),
+        |       (oliver:Person {name: 'Oliver Stone'}),
+        |       (rob:Person {name: 'Rob Reiner'}),
+        |
+        |       (wallStreet:Movie {title: 'Wall Street'}),
+        |       (charlie)-[:ACTED_IN {role: "Bud Fox"}]->(wallStreet),
+        |       (martin)-[:ACTED_IN {role: "Carl Fox"}]->(wallStreet),
+        |       (michael)-[:ACTED_IN {role: "Gordon Gekko"}]->(wallStreet),
+        |       (oliver)-[:DIRECTED]->(wallStreet),
+        |
+        |       (thePresident:Movie {title: 'The American President'}),
+        |       (martin)-[:ACTED_IN {role: "A.J. MacInerney"}]->(thePresident),
+        |       (michael)-[:ACTED_IN {role: "President Andrew Shepherd"}]->(thePresident),
+        |       (rob)-[:DIRECTED]->(thePresident)"""
     )
-  }
-
-  @Test def get_multiple_nodes() {
-    testQuery(
-      title = "Multiple starting points",
-      text = "By specifying two node patterns all combinations of the two nodes will be returned, so it is advised to add a constraint.",
-      queryText = """match (a), (b) where id(a) = id(b) return a, b""",
-      optionalResultExplanation = "Returns the combinations of the two nodes from the database.",
-      assertions = (p) => {
-        val allNodes: List[Node] = GlobalGraphOperations.at(db).getAllNodes.asScala.toList
-
-        assertEquals(allNodes, p.columnAs[Node]("a").toList)
+    synopsis("The `MATCH` clause is used to search for the pattern described in it.")
+    section("Introduction") {
+      p( """The `MATCH` clause allows you to specify the patterns Neo4j will search for in the database.
+           |This is the primary way of getting data into the current set of bindings.
+           |It is worth reading up more on the specification of the patterns themselves in <<introduction-pattern>>.""")
+      p( """`MATCH` is often coupled to a `WHERE` part which adds restrictions, or predicates, to the `MATCH` patterns, making them more specific.
+           |The predicates are part of the pattern description, not a filter applied after the matching is done.
```

Comment:
I believe the intention of this sentence is to emphasise the coupling of MATCH and WHERE, because the exact application of the WHERE depends on the logical plan, not on the Cypher syntax. So perhaps the WHERE is used in the leaf plan, perhaps in a post match filter, but that is a question of plan optimization, not of the Cypher syntax. I suggest we keep this section the same for now, and only deal with these subtleties in the section on query plan descriptions. Should we make a link to that section here?